### ISSUE TITLE
VRS performance evaluation

### ISSUE THREAD

**Issue by torbjorgen** [Clinical-Genomics/rdds#119]
Tag v1.7.0 provided a good model that performs decently on the MUTACC dataset.
Look deeper into the performance and test with some real data to observe model behavior.

## Tooling to introspect model data, inferences
Need to build tooling to be able to pinpoint characteristics of FN, FP samples contrasting the TP, TNs.

> Why are these samples wrongly predicted?

### Methods
 - [ ] Comparative analysis TN, TPs to FN, FPs (e.g what sets apart a TN to a FN)
  - [ ] Clustering?
 - Confirmative analysis whether FNs are out of sample to TPs (FNs are belong to same set as TPs)?

Above methods require that the text features are converted into a numerical representation.

 - XAI frameworks
  - https://github.com/shap/shap (seems most plausible w.r.t software compatibility)
    https://shap.readthedocs.io/en/latest/example_notebooks/tabular_examples/model_agnostic/Census%20income%20classification%20with%20scikit-learn.html
  - DEEPLIFT https://arxiv.org/pdf/1704.02685
  - DeepExplain https://github.com/marcoancona/DeepExplain
  - https://github.com/interpretml/interpret (lot's of reference readings)
  - https://github.com/tensorflow/lucid (Just TF 1.0, lots of reference readings)
    - https://github.com/pokman/mini_lucid_tf2 TF2 variant ?
 - ~Feature cross validation (but we want to have sample tests, not feature tests)~

https://distill.pub/2018/building-blocks/

#### Technical Solutions
- [ ] Use dataset pipeline text processing, embeddings to transform text data to numericals
  This implies to push FNs through the processing pipeline again?? No can we extract this side by side when running predictions?

### Test set, threshold 0.5 (suggests 0.2245 threshold, MCC)
- 171 FNs
- 32 FPs
  - 9/32 is considered possibly or probably damaging by VEP
  - 9/32 are frameshift variants (out of set compared to damaging above)
  - 6/32 are stop_gained variants
  - 4/32 are splice variants
  - 2 novel inframe insertion compared to Swegen frq
with threshold .5

Reanalysing with threshold 0.2224 (best MCC score gives):

    48 FPs, 50 % increase in FPs
    97 FNs, 55% reduction in FNs

## FP MUTACC Investigation
I'd say majority of the samples considered FP are in some way marked as pathogenic by at least one annotation.

Is this observed because of the wrong phenotype, clinicians considering something else?

## FN MUTACC Investigation

Judging from the model performance, some MUTACC samples are FNs, possibly due to missing CLINVAR annotations `CLNSIG` field. All FNs have missing data points.
This implies clinvara dataset overfitting.

Possible solutions going forward
 - [ ] Add more non-clinvar T data
 - [ ] regularisation
 - [ ] Dropout of clinvar CLNSIG field

- [ ] Visualize data properties Frq etc in train set vs mutacc validation set.

**Comment by torbjorgen** [comment]
Duplicates #111

**Comment by torbjorgen** [comment]
Closed as of PR #189 and #188
